community/kernel/src/main/java/org/neo4j/kernel/api/ProcedureRead.java

```diff
@@ -33,6 +34,9 @@
     /** Fetch a procedure given its signature. */
     ProcedureSignature procedureGet( ProcedureSignature.ProcedureName name ) throws ProcedureException;
 
+    /** List all procedures */
+    Iterator<ProcedureSignature> proceduresGetAll();
```

Comment:
They are unique, trying to register the same signature multiple times will throw. I think having it return a `Set` makes sense.